integration-tests/src/stream.rs

```
@@ -1,17 +1,20 @@
 #![cfg(test)]
 
 use bytes::Bytes;
-use lzma::LzmaError;
 use std::convert::Infallible;
-
+use std::io::Error;
+use std::io::Read;
 use test_context::futures::{stream, Stream};
+use xz2::read::XzDecoder;
 
 /// Take a slice, xa decompress the data and return it as a stream.
 ///
 /// If the decompression fails, that will result in an error on the stream.
-pub fn xz_stream(data: &[u8]) -> impl Stream<Item = Result<Bytes, LzmaError>> {
-    let result = lzma::decompress(data).map(|data| data.into());
-    stream::once(async { result })
+pub fn xz_stream(data: &[u8]) -> impl Stream<Item = Result<Bytes, Error>> {
+    let mut result = XzDecoder::new(data);
+    let mut extracted_content = Vec::new();
+    let _ = result.read_to_end(&mut extracted_content);
```

I don't think we should ignore the error.